sdk/client-core/core/src/main/java/io/clientcore/core/http/models/HttpRequest.java

```diff
@@ -178,60 +191,70 @@ public HttpRequest setBody(BinaryData content) {
     }
 
     /**
-     * Get the request metadata.
+     * Get the request {@link RequestOptions options}.
      *
-     * @return The request metadata.
+     * @return The request {@link RequestOptions options}.
      */
-    public HttpRequestMetadata getMetadata() {
-        return metadata;
+    public RequestOptions getOptions() {
+        return options;
     }
 
     /**
-     * Set the request metadata.
+     * Set the request {@link RequestOptions options}.
      *
-     * @param metadata The request metadata.
+     * @param options The request {@link RequestOptions options}.
      *
      * @return The updated {@link HttpRequest}.
      */
-    public HttpRequest setMetadata(HttpRequestMetadata metadata) {
-        Objects.requireNonNull(metadata, "'metadata' cannot be null");
+    public HttpRequest setOptions(RequestOptions options) {
+        Objects.requireNonNull(options, "'options' cannot be null");
 
-        this.metadata = metadata;
+        this.options = options;
 
         return this;
     }
 
     /**
-     * Creates a copy of the request.
-     * <p>
-     * The main purpose of this is so that this HttpRequest can be changed and the resulting HttpRequest can be a
-     * backup. This means that the cloned Headers and body must not be able to change from side effects of this
-     * HttpRequest.
+     * Get the specified event stream {@link ServerSentEventListener listener} for this request.
      *
-     * @return A new HTTP request instance with cloned instances of all mutable properties.
-     */
-    public HttpRequest copy() {
-        return new HttpRequest(httpMethod, url)
-            .setHeaders(new HttpHeaders(headers))
-            .setBody(body)
-            .setMetadata(metadata.copy());
-    }
-
-    /**
-     * Get the specified event stream listener for this request.
-     * @return the listener for this request.
+     * @return The {@link ServerSentEventListener listener} for this request.
      */
     public ServerSentEventListener getServerSentEventListener() {
         return serverSentEventListener;
     }
 
     /**
-     * Set an event stream listener for this request.
-     * @param serverSentEventListener the listener to set for this request.
-     * @return This HttpRequest.
+     * Set an event stream {@link ServerSentEventListener listener} for this request.
+     *
+     * @param serverSentEventListener The {@link ServerSentEventListener listener} to set for this request.
+     *
+     * @return The updated {@link HttpRequest}.
      */
     public HttpRequest setServerSentEventListener(ServerSentEventListener serverSentEventListener) {
         this.serverSentEventListener = serverSentEventListener;
+
+        return this;
+    }
+
+    /**
+     * Gets the number of times the request has been retried.
+     *
+     * @return The number of times the request has been retried.
+     */
+    public int getRetryCount() {
```

Comment:
Do we need this to be public? Can we make this private similar to the setter?

Reply:
Yes, we can make it private.